src/tty/unix.rs

```diff
@@ -979,11 +979,21 @@ impl Renderer for PosixRenderer {
         }
         // display hint
         if let Some(hint) = hint {
+            // position the cursor within the line
```

Comment:
So your PR works only on unix, right ?

Reply:
I do not have access to any Windows. I also don't know if Windows terminals have an insert mode? The resulting difference between Unix and Windows is that for Windows the hint is still displayed at the end of the line while it is displayed right next to the cursor on Unix. So, the difference is on presentation level only.

Reply:
See https://github.com/kkawakam/rustyline#wine